rackcity/permissions/permissions.py

```diff
@@ -75,48 +75,90 @@ def get_permission(permission_name: PermissionName) -> Permission:
     return permission
 
 
-def user_has_asset_permission(user, site=None):
-    if user.has_perm(PermissionPath.ASSET_WRITE.value):
-        return True
-    if site:
+def user_has_asset_permission(user, site):
+    return user.has_perm(
+        PermissionPath.ASSET_WRITE.value
+    ) or site.user_has_site_level_permission(user)
+
+
+def get_id_from_data(data, field, data_is_validated):
+    if data_is_validated:
+        return data[field].id
+    else:
+        return data[field]
```

Comment:
added this because we call it from asset_add with serializer validated data, which serializes foreign keys to their models so you need to access id. but we also call it from asset_modify with the straight up data that is sent so foreign keys are listed as their id.